### ISSUE TITLE
How to improve my model and compare(YOUR_MODEL vs MY_MODEL)

### ISSUE THREAD

**Issue by sam-x-specx** [campusx-official/100-days-of-deep-learning#8]
from sklearn.base import BaseEstimator, ClassifierMixin

class Perceptron_model(BaseEstimator, ClassifierMixin):
    def __init__(self, learning_rate=0.1, n_iters=1000):
        self.learning_rate = learning_rate
        self.n_iters = n_iters
        self.weights = None
        self.bias = None

    def stepfun(self ,X):
            return np.where(X >= 0 ,1 ,0)
    
    def fit(self, X, y):
        # Add bias term as first column
        X = np.insert(X, 0, 1, axis=1)  # bias column
        n_samples, n_features = X.shape
        self.weights = np.ones(n_features)
    
        for epoch in range(self.n_iters):
            indices = np.arange(n_samples)
            np.random.shuffle(indices)
    
            for idx in indices:
                x_i = X[idx]
                y_true = y[idx]
                z = np.dot(x_i, self.weights)
                y_pred = self.stepfun(z)
                update = self.learning_rate * (y_true - y_pred)
                self.weights += update * x_i
    
        return self
    
    def predict(self, X):
        X = np.insert(X, 0, 1, axis=1)
        linear_output = np.dot(X, self.weights)
        return self.stepfun(linear_output)

I compare your perceptron vs my perceptron model and find less accuracy how to improve my model by my way
